File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.js
const solution = (array) => {
    let dict = {}
    for (v of array) {
        if (!dict[v]) dict[v]=1
        else dict[v]++
    }
    let max = 0 // 최대인 원소값
    let cnt = -1 // 최대수
    array.forEach(v => {
        if (!dict[max]) max = v 
        cnt = Math.max(dict[max],dict[v])
        if (dict[v]==cnt){
            max = v
        }
    }) 
    let lenArr = []
    for (i of array) {
        if (dict[i] === cnt) lenArr.push(i)
    } 
    let setArr = new Set(lenArr)
    if (setArr.size > 1) return -1
    return max
    
}

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.py
def solution(array):
    dic = {}   
    for i in array:
        dic[i] = dic.get(i, 0) +1
    val = sorted(dic.values(), reverse=True)
    max = val[0]  
    print(len(dic))
    if len(dic)==1: return array[0]
    if val[1] == max: return -1 
    for i in dic:
        if dic[i] == max: 
            print(i)
            return i
     